### ISSUE TITLE
ksp throwing build error when using @Retryable 

### ISSUE THREAD

**Issue by vensad** [micronaut-projects/micronaut-core#9727]
### Expected Behavior

Using Retryable annotation on a fun within singleton class , it produces compilation errors
when executing gradle build , it should complete normally w/o any error 

```
./gradlew.bat  build 
```

### Actual Behaviour

it produces error related to ksp and AOP 
```
 ./gradlew.bat build 

Welcome to Gradle 8.2.1!

Here are the highlights of this release:
 - Kotlin DSL: new reference documentation, assignment syntax by default
 - Kotlin DSL is now the default with Gradle init
 - Improved suggestions to resolve errors in console output
 - Reduced sync memory consumption

For more details see https://docs.gradle.org/8.2.1/release-notes.html

Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details

> Task :kspKotlin FAILED
e: [ksp] Originating element: getContent
e: [ksp] C:/GitHub/demo/src/main/kotlin/io/test/service/MicronautService.kt:9: Method defines AOP advice but is declared final. Change the method to be non-final in order for AOP advice to be applied.
e: Error occurred in KSP, check log for detail
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 11s
4 actionable tasks: 1 executed, 3 up-to-date
```

### Steps To Reproduce

Try executing gradle build by using one of the below 
```
./gradlew build

```
or 
```
./gradlew.bat build 
```

### Environment Information

Windows/Linux/MacOS
JDK18 (same issue with JDK17)
Micronaut4 with ksp plugin 
Kotlin 1.8 

### Example Application

https://github.com/vensad/demo

### Version

4.0.3

**Comment by dstepanov** [comment]
Have you tried to make it open?

**Comment by vensad** [comment]
@dstepanov it is working by making it open, but i don't think i can do the same for existing services with 3.x to make them use micronaut4. More of it seems to issue with AOP not specific to Retryable.

**Comment by vensad** [comment]
below solutions work great too , but i have to annotate it class level . 
```
@Singleton
@InterceptorBinding
class MicronautService() {
    @Retryable
    suspend fun getContent(): String{
        return ""
    }
}
```
```
@Singleton
@Retryable
class MicronautService() {
    suspend fun getContent(): String{
        return ""
    }
}

**Comment by dstepanov** [comment]
The open plugin is not compatible with KSP, we have some "hacks" to make it work, but it looks like there are some missing cases.

**Comment by vensad** [comment]
yup its one of such when i am trying to apply it on function level but not class. https://github.com/micronaut-projects/micronaut-core/pull/9456 helped upto some extent by solving this issue at class level , i do run into issues with function level.

**Comment by vensad** [comment]
Do you suggest any changes to [this](https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/inject-kotlin/src/main/kotlin/io/micronaut/kotlin/processing/visitor/AbstractKotlinElement.kt#L121)  to consider function level also ?

**Comment by dstepanov** [comment]
@yawkat Any idea why is it not open?

**Comment by yawkat** [comment]
we only detect annotations at class level, because that's what i thought kotlin did. but i just checked, and for the k2 compiler indeed it seems to only check class level, but for the k1 compiler the all-open plugin considers method level annotations too.

**Comment by graemerocher** [comment]
Triage: validated KSP/AOP processor issue, with an external reproducer and a current-source indication that method-level openness is still missed. `AbstractKotlinElement.isFinal()` calls `shouldBeOpen(owningType)` for members, so a method annotated only with `@Retryable` is not treated like a class annotated with the same advice. The workaround remains making the method/class `open` or placing the advice at class level. Suggested fix: evaluate the member’s merged annotation metadata for Micronaut AOP stereotypes while preserving Kotlin all-open configuration, then add KSP tests for method-level `@Retryable` under K1/K2 and class-level control cases.
